pennylane/tape/interfaces/autograd.py

```diff
@@ -195,18 +195,67 @@ def vjp(ans, self, params, device):  # pylint: disable=unused-argument
             gradient output vector, and computes the vector-Jacobian product
         """
 
-        def gradient_product(g):
-            # In autograd, the forward pass is always performed prior to the backwards
-            # pass, so we do not need to re-unwrap the parameters.
+        saved_grad_matrices = {}

```

Comment:
Not a big fan of a global storing solution. This is going to kill memory usage if they users runs a ton of experiments.

Reply:
Good point. The idea here is to simply cache the hessian/jacobian for the _same_ parameter value, if `vjp` is called repeatedly by Autograd. So repeated calls by the user via `qml.jacobian` won't do any caching, but a single call by the user to `qml.jacobian` will cache under the hood for autograd.

Is there anyway to cache while ensuring 'local' scope?

Ideally I need some sort of dynamic programming for Python